AK/Variant.h

```diff
@@ -40,47 +40,137 @@ consteval IndexType index_of()
     return VariantIndexOf<T, IndexType, 0, Ts...> {}();
 }
 
+template<typename... T>
+union VariantStorage;
+
+template<typename T, typename... Rest>
+union VariantStorage<T, Rest...> {
+    VariantStorage() = default;
+    VariantStorage()
+    requires(!IsTriviallyConstructible<T> || (!IsTriviallyConstructible<Rest> || ...)) {};
+
+    VariantStorage(VariantStorage&&) = default;
+    VariantStorage(VariantStorage const&) = default;
+
+    VariantStorage& operator=(VariantStorage&&) = default;
+    VariantStorage& operator=(VariantStorage const&) = default;
+
+    // Note: Actual destruction is handled by the Variant itself
+    ~VariantStorage() = default;
+    ~VariantStorage()
+    requires(!IsTriviallyDestructible<T> || !(IsTriviallyDestructible<Rest> && ...)) {};
+    ~VariantStorage()
+    requires(!IsDestructible<T> || !(IsDestructible<Rest> && ...))
+    = delete;
+
+    VariantStorage<Rest...> rest;
+    T value;
+};
+
+template<>
+union VariantStorage<> {
+};
+
 template<typename IndexType, IndexType InitialIndex, typename... Ts>
-struct Variant;
+struct VariantHelper;
 
 template<typename IndexType, IndexType InitialIndex, typename F, typename... Ts>
-struct Variant<IndexType, InitialIndex, F, Ts...> {
+struct VariantHelper<IndexType, InitialIndex, F, Ts...> {
     static constexpr auto current_index = VariantIndexOf<F, IndexType, InitialIndex, F, Ts...> {}();
-    ALWAYS_INLINE static void delete_(IndexType id, void* data)
+
+    constexpr static void delete_(IndexType state, VariantStorage<F, Ts...>& storage)
```

Comment:
with all this de-inlining, please present perf change reports.
Any benchmark should do, variant is sufficiently integrated I think.

Reply:
Sure, will put ALWAYS_INLINE back, although I think that that is implied by constexpr

Reply:
`inline` is implied (on functions), the attribute is not.